filesystems/memory.py

```diff
@@ -48,7 +48,7 @@ class _File(object):
     def __getitem__(self, name):
         return _FileChild(parent=self._parent)
 
-    def create_directory(self, path):
+    def create_directory(self, path, parents=False):
```

Comment:
Would probably try to avoid repeating the default value -- this "API" is internal to these objects, so you can even make parents be required to provide at this layer.